framework/aster-frame/src/arch/x86/tdx_guest.rs

```diff
@@ -87,3 +150,279 @@ fn handle_io(trapframe: &mut dyn TdxTrapFrame, ve_info: &tdcall::TdgVeInfo) -> b
     };
     true
 }
+
+fn is_shared_gpa(gpa: u64) -> bool {
```

Comment:
One thing that is really confusing to me is that if the shared bit should be consider part of the GPA? This method seems to suggest this is the case. But our `unprotect_phys_pages` method also takes a `gpa` as the starting address. But that `gap` is not supposed to contain the shared bit. I think we use the concept of GPAs consistently in our implementation.